master/buildbot/www/config.py

```diff
@@ -39,11 +41,51 @@ def reconfigResource(self, new_config):
         if isinstance(self.config.get('versions'), list):
             versions += self.config['versions']
 
+        template_dir = self.config.get('custom_templates_dir')
+        if template_dir is not None:
+            del self.config['custom_templates_dir']
+            template_dir = os.path.join(self.master.basedir, template_dir)
+            self.custom_templates = self.parseCustomTemplateDir(template_dir)
+        else:
+            self.custom_templates = {}
         self.config['versions'] = versions
 
     def render_GET(self, request):
         return self.asyncRenderHelper(request, self.renderIndex)
 
+    def parseCustomTemplateDir(self, template_dir):
+        res = {}
+        allowed_ext = [".html"]
+        try:
+            import pyjade
+            allowed_ext.append(".jade")
+        except ImportError:  # pragma: no cover
+            log.msg("pyjade not installed. Ignoring .jade files from %s" % (template_dir,))
+            pyjade = None
+        for root, dirs, files in os.walk(template_dir):
+            if root == template_dir:
+                template_name = "views/%s.html"
```

Comment:
I'm not 100% sure here, but `os.path.join('views', '%s.html')` might be better here.